x/staking/types/msg.go

```diff
@@ -421,3 +412,198 @@ func (m *MsgUpdateParams) GetSigners() []sdk.AccAddress {
 	addr, _ := sdk.AccAddressFromBech32(m.Authority)
 	return []sdk.AccAddress{addr}
 }
+
+// NewMsgTokenizeShares creates a new MsgTokenizeShares instance.
+//
+//nolint:interfacer
+func NewMsgTokenizeShares(delAddr sdk.AccAddress, valAddr sdk.ValAddress, amount sdk.Coin, shareOwner sdk.AccAddress) *MsgTokenizeShares {
+	return &MsgTokenizeShares{
+		DelegatorAddress:    delAddr.String(),
+		ValidatorAddress:    valAddr.String(),
+		Amount:              amount,
+		TokenizedShareOwner: shareOwner.String(),
+	}
+}
+
+// Type implements the sdk.Msg interface.
+func (msg MsgTokenizeShares) Type() string { return TypeMsgTokenizeShares }
+
+func (msg MsgTokenizeShares) GetSigners() []sdk.AccAddress {
+	delegator, err := sdk.AccAddressFromBech32(msg.DelegatorAddress)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{delegator}
+}
+
+func (msg MsgTokenizeShares) GetSignBytes() []byte {
+	bz := legacy.Cdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg MsgTokenizeShares) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.DelegatorAddress); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid delegator address: %s", err)
+	}
+	if _, err := sdk.ValAddressFromBech32(msg.ValidatorAddress); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid validator address: %s", err)
+	}
+	if _, err := sdk.AccAddressFromBech32(msg.TokenizedShareOwner); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid tokenize share owner address: %s", err)
+	}
+
+	if !msg.Amount.IsValid() || !msg.Amount.Amount.IsPositive() {
+		return sdkerrors.Wrap(
+			sdkerrors.ErrInvalidRequest,
+			"invalid shares amount",
+		)
+	}
+
+	return nil
+}
+
+// NewMsgRedeemTokensforShares creates a new MsgRedeemTokensforShares instance.
+//
+//nolint:interfacer
+func NewMsgRedeemTokensforShares(delAddr sdk.AccAddress, valAddr sdk.ValAddress, amount sdk.Coin, shareOwner sdk.AccAddress) *MsgRedeemTokens {
+	return &MsgRedeemTokens{
+		DelegatorAddress: delAddr.String(),
+		Amount:           amount,
+	}
+}
+
+// Type implements the sdk.Msg interface.
+func (msg MsgRedeemTokens) Type() string { return TypeMsgRedeemTokens }
+
+func (msg MsgRedeemTokens) GetSigners() []sdk.AccAddress {
+	delegator, err := sdk.AccAddressFromBech32(msg.DelegatorAddress)
```

Comment:
```suggestion
	delegator := sdk.MustAccAddressFromBech32(msg.DelegatorAddress)
```

When we must panic, we can use the helper `MustAccAddressFromBech32` instead.